src/data-table/data-table.tsx

```diff
@@ -60,15 +150,64 @@ export const DataTable = ({
         return headers.length + (enableMultiSelect ? 1 : 0);
     };
 
+    const calculateFixedInViewport = () => {
+        if (!wrapperRef.current) {
+            return;
+        }
+        const wrapperBounds = wrapperRef.current.getBoundingClientRect();
+        setIsFloatingActionBar(
+            wrapperBounds.bottom > window.innerHeight + 30 &&
+                wrapperBounds.top < window.innerHeight - 200
+        );
+    };
+
+    const calculateStickyInViewport = () => {
+        if (
+            !tableRef.current ||
+            !tableEndRef.current ||
+            !wrapperRef.current ||
+            !actionBarRef.current ||
+            !headerRef.current
+        ) {
+            return;
+        }
+
+        const endBounds = tableEndRef.current.getBoundingClientRect();
+
+        if (endBounds.top > window.innerHeight) {
+            const bottomOffset = endBounds.bottom - window.innerHeight;
+            const bottomToHeaderOffset =
+                tableRef.current.getBoundingClientRect().height -
+                headerRef.current.clientHeight -
+                32;
+            const maxBottomOffset = Math.min(
+                bottomOffset,
+                bottomToHeaderOffset
+            );
+
+            actionBarRef.current.style.transform = `translateY(-${maxBottomOffset}px)`;
+        } else {
+            actionBarRef.current.style.transform = `translateY(0)`;
+        }
+    };
+
+    const checkLastBorder = () => {
+        setShowLastBorder(
+            tableRef.current?.scrollHeight <
+                (tableRef.current?.childNodes[0] as HTMLElement).clientHeight
+        );
```

Comment:
in which scenario is the table scrollHeight less than the height of the first row?